clickhouse-core/src/testFixtures/scala/xenon/clickhouse/base/ClickHouseClusterMixIn.scala

```diff
@@ -24,14 +24,20 @@ import java.io.File
 
 trait ClickHouseClusterMixIn extends AnyFunSuite with ForAllTestContainer {
 
+  val CLICKHOUSE_VERSION: String = Utils.load("CLICKHOUSE_VERSION", "23.8")
+  val isCloud: Boolean = if (CLICKHOUSE_VERSION.equalsIgnoreCase("cloud")) true else false
+  val isOnPrem: Boolean = !isCloud
+  val CLICKHOUSE_IMAGE: String = if (isCloud) "clickhouse/clickhouse-server:23.8"
```

Comment:
this is too hacky.

I would suggest adding a new `ClickHouseCouldMixIn` alongside the existing `ClickHouseSingleMixIn` and `ClickHouseClusterMixIn`.

Reply:
this comment was ignored. the current change mixes things up. Let me clarify more details

`ClickHouse*MixIn` tasks responsibility to provide a ClickHouse instance(s) to run integration tests:

- `ClickHouseSingleMixIn` provides a single ClickHouse instance by using docker to launch a single container
- `ClickHouseClusterMixIn` provides a ClickHouse cluster by using docker compose to launcher a set of containers
- `ClickHouseCouldMixIn` should just expose ClickHouse Cloud instance's URL and credential.

Reply:
this will introduce some refactoring, i will take a look if i find time this weekend